src/GraphQL.Tests/Subscription/SubscriptionExecutionStrategyTests.cs

```diff
@@ -0,0 +1,678 @@
+        Observer.ShouldHaveResult().ShouldBeSimilarTo(@"{ ""data"": { ""testComplex"": { ""id"": ""SampleId"", ""nameMayThrowError"": ""testing"" } } }");
+        Observer.ShouldHaveResult().ShouldBeSimilarTo(@"{""errors"":[{""message"":""Error trying to resolve field \u0027nameMayThrowError\u0027."",""locations"":[{""line"":1,""column"":33}],""path"":[""testComplex"",""nameMayThrowError""],""extensions"":{""code"":""APPLICATION"",""codes"":[""APPLICATION""]}}],""data"":null}");
+        Observer.ShouldHaveResult().ShouldBeSimilarTo(@"{ ""data"": { ""testComplex"": { ""id"": ""SampleId"", ""nameMayThrowError"": ""success"" } } }");
+        Observer.ShouldHaveNoMoreResults();
+    }
+
+    [Fact]
+    public async Task Widget_WithErrors_Nullable()
+    {
+        // a field error has occurred on a nullable field, so just that field is null
+        var result = await ExecuteAsync("subscription { testComplex { id nameMayThrowErrorNullable } }");
+        result.ShouldBeSuccessful();
+        Source.Next("hello");
+        Source.Next("custom");
+        Source.Next("testing");
+        Source.Next("application");
+        Source.Next("success");
+        Observer.ShouldHaveResult().ShouldBeSimilarTo(@"{ ""data"": { ""testComplex"": { ""id"": ""SampleId"", ""nameMayThrowErrorNullable"": ""hello"" } } }");
+        Observer.ShouldHaveResult().ShouldBeSimilarTo(@"{""errors"":[{""message"":""Handled custom exception: Custom error"",""locations"":[{""line"":1,""column"":33}],""path"":[""testComplex"",""nameMayThrowErrorNullable""],""extensions"":{""code"":""INVALID_OPERATION"",""codes"":[""INVALID_OPERATION""]}}],""data"":{""testComplex"":{""id"":""SampleId"",""nameMayThrowErrorNullable"":null}}}");
+        Observer.ShouldHaveResult().ShouldBeSimilarTo(@"{ ""data"": { ""testComplex"": { ""id"": ""SampleId"", ""nameMayThrowErrorNullable"": ""testing"" } } }");
+        Observer.ShouldHaveResult().ShouldBeSimilarTo(@"{""errors"":[{""message"":""Error trying to resolve field \u0027nameMayThrowErrorNullable\u0027."",""locations"":[{""line"":1,""column"":33}],""path"":[""testComplex"",""nameMayThrowErrorNullable""],""extensions"":{""code"":""APPLICATION"",""codes"":[""APPLICATION""]}}],""data"":{""testComplex"":{""id"":""SampleId"",""nameMayThrowErrorNullable"":null}}}");
+        Observer.ShouldHaveResult().ShouldBeSimilarTo(@"{ ""data"": { ""testComplex"": { ""id"": ""SampleId"", ""nameMayThrowErrorNullable"": ""success"" } } }");
+        Observer.ShouldHaveNoMoreResults();
+    }
+
+    [Fact]
+    public async Task Widget_SourceError()
+    {
+        // in this case the graphql execution has not started, so executed should be false
+        // and "data" should not exist in the map of error events
```

Comment:
Do these comments accurately belong to this test?

Reply:
yes, the comments are accurate.  they refer to the event stream resolvers' executions, and the tests in the bottom of the method verify that 'data' is not in the returned map.

Reply:
```suggestion
        // during handling of the `Source.Error` events below,
        // in this case the graphql execution has not started, so executed should be false
        // and "data" should not exist in the map of error events
```

Reply:
Comment updated as shown above.